web3.js/src/connection.ts

```diff
@@ -3000,6 +3034,7 @@ export class Connection {
   async confirmTransaction(
     strategy:
       | BlockheightBasedTransactionConfirmationStrategy
+      | NonceBasedTransactionConfirmationStrategy
```

Comment:
This method has gotten very complex. Rather than increase the complexity here I think we should probably add a new `confirmNonceTransaction` method or break out some of the internal logic into separate functions. At a high level, transaction confirmation has two concurrent processes which detect confirmation and expiration, respectively. Confirmation looks the same across all strategies so it really looks like we have different handling for detecting expiration. Maybe each expiration strategy gets its own function?

Reply:
So, yes, 100%. I plan on making each strategy it’s own function in the rewrite of this jalopy. As for this version of the library, I think it’s too late.

breaking it up into multiple functions will:

* confuse people

while not actually yielding any of the benefits of breaking them up:

* won’t reduce code size, because all of the functions are class methods

I can for sure break the code up internally, to make it easier for us framework devs to follow, but there are definitely going to be things about that you won’t like. For instance, I’m going to have to pass around the `done` flag as a mutable ref, which is going to be super gross.

Reply:
I personally think that overloading functions is more confusing, but I'm sure different people are confused by different things. I'm not concerned about code size, more concerned with maintainability of this code. If you feel strongly about keeping this overloading approach, feel free to keep it that way but I'd prefer methods to be explicit. Something to consider is that we may want to mimic this useful API on Rust which we likely would split up into separate public functions.

Reply:
I 100% agree that these should be different methods (and different modules!). It's just too late for this API; we already did the block height strategy using an overload so it's too late to turn back now.

Until the rewrite of course :)